Laura dev

lpetrich opened the pull request:
updated visualizer with flags so we can switch to plotting only emg or only imu

ShriramRaghu-UofA committed:
Update emg_predictor.py

Improve shared memory tracking for online streamer. Previously we didn't have a way of knowing when the shared memory was updated. Similar to other parts of the code, introducing a _count tag should help alleviate that issue. Note that it appears that `classifier_smm_writes` was being used for something similar, but given that it was initialized as process-local variable, this won't be reflected back to the parent if not running in blocking mode.

ShriramRaghu-UofA committed:
Update emg_predictor.py

Previous versions of libEMG created socket objects prior to process spawn which generally leads to errors. Create the sockets in the main loop instead.

lpetrich committed:
Shri: Fix locking issue with libemg implementation.

lpetrich committed:
end of quinn testing working scripts

lpetrich committed:
update gitignore with files that wont go away

lpetrich committed:
added flag to visualizer to plot emg and or imu

ShriramRaghu-UofA commented on libemg/_datasets/full_reachable_workspace.py:
We can revert this now that Recorder uses comma by default

lpetrich replied:
done

ShriramRaghu-UofA commented on libemg/data_handler.py:
```diff
@@ -890,7 +890,7 @@ def update(frame):
 
 
-    def visualize_channels(self, channels, num_samples=500, y_axes=None):
+    def visualize_channels(self, channels, num_samples=500, y_axes=None, plot_emg=True, plot_imu=False):
```
This is a good quick way to get a more readable plot!

Good default is to perhaps only plot EMG unless we really need to see the IMU. 

However, this code:

```python
fig, ax = pyplot.subplots(2, 1, figsize=(12, 6), height_ratios=[1, 9])
```
Will make the plot still unreadable since we will have a blank space below that won't be used. There are two options here:

First, don't even instantiate the second axis:

```python
fig, ax = pyplot.subplots(1, 1, figsize=(12, 6))
```
However, this will lead to a lot of work for you since ax is no longer a list, so you will have to update all the code to change it. Alternatively, a quick hack could be to do:

```python
fig, ax = pyplot.subplots(2, 1, figsize=(12, 6), height_ratios=[20, 1])
```

Or something like that so that we use max space for EMG. This will leave a small space below for the "IMU" plot but you won't have to change most other stuff. Pick whichever is convenient but for the long term, we should do it properly and perhaps separate out EMG and IMU plotting on to it's own private methods which takes in an ax object. That way, we can dispatch the method if we only need it.

lpetrich replied:
Made the changes, should fix better in the future

ShriramRaghu-UofA reviewed (changes requested):
Plotting requires changes to make it more visible.

lpetrich committed:
Update full_reachable_workspace.py

reverted back to comma delimiter

lpetrich committed:
Update data_handler.py

handled sizing of plots as suggested by shri should make more robust in the future

lpetrich committed:
Merge branch 'main' into laura-dev
